File: solutions/ruby/raindrops/1/raindrops.rb
module BookKeeping
  VERSION = 3
end

class Raindrops
  @table = { 3 => "Pling", 5 => "Plang", 7 => "Plong" }
  def self.convert(num)
    result = ""
    @table.each do |key, word|
      if num % key == 0
        result << word
      end
    end
    if result.empty?
      return num.to_s
    end
    return result
  end
end

File: solutions/go/raindrops/1/raindrops.go
package raindrops

import (
	"strconv"
	"strings"
)

func Convert(number int) string {
	s := strings.Builder{}
	if number%3 == 0 {
		s.WriteString("Pling")
	}
	if number%5 == 0 {
		s.WriteString("Plang")
	}
	if number%7 == 0 {
		s.WriteString("Plong")
	}
	if s.Len() == 0 {
		s.WriteString(strconv.Itoa(number))
	}
	return s.String()
}
